collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -595,12 +598,23 @@ protected void onActivityResult(int requestCode, int resultCode,
             return;
         }
 
-        switch (requestCode) {
-            case BARCODE_CAPTURE:
+        // For handling results returned by the Zxing Barcode scanning library
+        IntentResult barcodeScannerResult = IntentIntegrator.parseActivityResult(requestCode, resultCode, intent);
+        if (barcodeScannerResult != null) {
+            if (barcodeScannerResult.getContents() == null) {
+                // request was canceled...
+                Timber.i("QR code scanning cancelled");
+            } else {
                 String sb = intent.getStringExtra("SCAN_RESULT");
```

Comment:
@shobhitagarwal1612 I actually wrote both statements in the same place just for comparison..i.e 
Line 1: String sb1 = barcodeScannerResult.getContents();
and 
Line 2: String sb2 = intent.getStringExtra("SCAN_RESULT");

Turns out the value stored in sb1 and sb2 are exactly identical.i.e both statements are storing the same value in sb. 

Would you still recommend replacing the current statement
String sb = intent.getStringExtra("SCAN_RESULT"); 
with
 String sb = barcodeScannerResult.getContents();
If yes, why?